Description: fixes segfault on Assistant's startup
 This patch calls the static method QApplication::allWidgets()
 intead of storing them first.
Author: Than Ngo <than@redhat.com>
Origin: https://bugreports.qt-project.org/browse/QTBUG-25324
Bug: https://bugreports.qt-project.org/browse/QTBUG-25324
Bug-Debian: http://bugs.debian.org/679874
Forwarded: not-needed
Last-Update: 2012-06-20

--- qt4-x11-4.8.2.orig/tools/assistant/tools/assistant/mainwindow.cpp
+++ qt4-x11-4.8.2/tools/assistant/tools/assistant/mainwindow.cpp
@@ -944,8 +944,7 @@ void MainWindow::updateApplicationFont()
     if (helpEngine.usesAppFont())
         font = helpEngine.appFont();
 
-    const QWidgetList &widgets = qApp->allWidgets();
-    foreach (QWidget* widget, widgets)
+    foreach (QWidget* widget, QApplication::allWidgets())
         widget->setFont(font);
 }
 
